rustdoc-json/src/build.rs

```diff
@@ -26,7 +27,9 @@ pub(crate) fn run_cargo_rustdoc(options: BuildOptions) -> Result<PathBuf, BuildE
         rustdoc_json_path_for_manifest_path(options.manifest_path, options.target.as_deref())
     } else {
         let manifest = cargo_toml::Manifest::from_path(&options.manifest_path)?;
-        if manifest.workspace.is_some() {
+        if manifest.package.is_some() {
+            Err(BuildError::RustDocError)
+        } else if manifest.workspace.is_some() {
             Err(BuildError::VirtualManifest(options.manifest_path))
         } else {
             Err(BuildError::General(String::from("See above")))
```

Comment:
Maybe this should actually be

```suggestion
        if manifest.package.is_none() && manifest.workspace.is_some() {
            Err(BuildError::VirtualManifest(options.manifest_path))
        } else {
            Err(BuildError::General(String::from("See above")))
```